pontoon/base/views.py

```diff
@@ -364,20 +367,80 @@ def contributors(request, template='users.html'):
         .exclude(string=F('entity__string_plural'))
     )
 
+    translations_total = (
+        translations.annotate(count=Count('user'))
+    )
+
+    translations_approved = (
+        translations.filter(approved=True).annotate(count=Count('user'))
+    )
+
+    translations_unapproved = (
+        translations.exclude(approved=True).exclude(fuzzy=True).annotate(count=Count('user'))
+    )
+
+    translations_needs_work = (
+        translations.filter(fuzzy=True).annotate(count=Count('user'))
+    )
+
     users = (
         User.objects
         .filter(translation__in=translations).distinct()
         .exclude(email__in=settings.EXCLUDE)
         .annotate(translation_count=Count('translation'))
         .exclude(translation_count=0)
-        .order_by('-translation_count')[:100]
     )
 
-    for user in users:
-        user.translations = translations.filter(user=user)
+    today = datetime.date.today()
+
+    end_date = datetime.datetime(today.year, today.month, 1)
+    start_date = None
+
+    if period == '12months':
+        start_date = end_date - relativedelta(month=+12)
+    elif period == '6months':
+        start_date = end_date - relativedelta(month=+6)
+    elif period == '3months':
+        start_date = end_date - relativedelta(month=+3)
+    elif period == 'lastmonth':
+        start_date = end_date - relativedelta(month=+1)
+
+    if start_date and end_date:
```

Comment:
If you filter the translation queryset before building the separate querysets and users list you can drop this block down to like 4 lines and avoid a bunch of extra filtering.